src/cli/cli.cpp

```diff
@@ -2480,10 +2480,75 @@ template <> otError Interpreter::Process<Cmd("counters")>(Arg aArgs[])
      */
     if (aArgs[0].IsEmpty())
     {
+#if OPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE
+        OutputLine("br");
+#endif
         OutputLine("ip");
         OutputLine("mac");
         OutputLine("mle");
     }
+#if OPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE
+    /**
+     * @cli counters br
+     * @code
+     * > counters br
+     * Inbound Unicast: Packets 4 Bytes 320
+     * Inbound Multicast: Packets 0 Bytes 0
+     * Outbound Unicast: Packets 2 Bytes 160
+     * Outbound Multicast: Packets 0 Bytes 0
+     * Done
+     * @endcode
+     * @cparam counters @ca{br}
+     * @par api_copy
+     * #otIp6GetBorderRoutingCounters
+     */
+    else if (aArgs[0] == "br")
+    {
+        if (aArgs[1].IsEmpty())
+        {
+            struct BrCounterName
+            {
+                const otPacketsAndBytes otBorderRoutingCounters::*mPacketsAndBytes;
+                const char *                                      mName;
+            };
+
+            static const BrCounterName kCounterNames[] = {
+                {&otBorderRoutingCounters::mInboundUnicast, "Inbound Unicast"},
+                {&otBorderRoutingCounters::mInboundMulticast, "Inbound Multicast"},
+                {&otBorderRoutingCounters::mOutboundUnicast, "Outbound Unicast"},
+                {&otBorderRoutingCounters::mOutboundMulticast, "Outbound Multicast"},
+            };
+
+            const otBorderRoutingCounters *brCounters = otIp6GetBorderRoutingCounters(GetInstancePtr());
+            Uint64StringBuffer             u64StringBuffer;
+
+            for (const BrCounterName &counter : kCounterNames)
+            {
+                OutputLine("%s: Packets %s Bytes %s", counter.mName,
+                           Uint64ToString((brCounters->*counter.mPacketsAndBytes).mPackets, u64StringBuffer),
+                           Uint64ToString((brCounters->*counter.mPacketsAndBytes).mBytes, u64StringBuffer));
```

Comment:
We cannot use the same `u64StringBuffer` twice in the same overall `OutputLine()` statement.
The second call will reuse the same underlying char buffer and overwrite the result of the first one.

We need to either define two buffers, or break the `OutputLine()` call (would suggest this). (Can also use `OutputUint64()`).

Reply:
I ended up using 2 buffers. I initially wanted to do it like: 
```
OutputFormat("%s ", ...);
OutputFormat(" Packets %s", ...);
OutputFormat(" Bytes %s", ...);
OutputLine("");
```

However, there was a [CI](https://github.com/openthread/openthread/actions/runs/3415652056/jobs/5685006317) complaining `OutputLine("")` passes an empty string to `gnu_printf`. I'm not sure why this was reported in this new PR since there are already several `OutputLine("")`s in the code. 

In a future PR, can I introduce a method `OutputLine()` which just does `OutputFormat("\r\n")`?

Reply:
We already have `OutputNewLine()` which does this. Added it for this purpose exactly from https://github.com/openthread/openthread/pull/8369. :) 

Yes. For some reason `gnu_printf` check does not like empty string passed with `%s`.